cmd/query/app/flags.go

```diff
@@ -36,20 +36,24 @@ import (
 )
 
 const (
+	queryHostPort           = "query.host-port"
 	queryPort               = "query.port"
 	queryBasePath           = "query.base-path"
 	queryStaticFiles        = "query.static-files"
 	queryUIConfig           = "query.ui-config"
 	queryTokenPropagation   = "query.bearer-token-propagation"
 	queryAdditionalHeaders  = "query.additional-headers"
 	queryMaxClockSkewAdjust = "query.max-clock-skew-adjustment"
+	queryPortWarning        = "(deprecated, will be removed after 2020-06-30 or in release v1.20.0, whichever is later)"
 )
 
 // QueryOptions holds configuration for query service
 type QueryOptions struct {
-	// Port is the port that the query service listens in on
+	// HostPort is the host:port address that the query service listens o n
+	HostPort string
+	// Port is the port that the query service listens in on ()
 	Port int
-	// BasePath is the prefix for all UI and API HTTP routes
+	// BasePath is the prefix for all UI and API HTTP routes (deprecated, will be removed after 2020-06-30 or in release v1.20.0, whichever is later)
```

Comment:
why is BasePath deprecated?

Reply:
To be honest that was a total brainfart from my side.^^